controller_interface/include/semantic_components/force_torque_sensor.hpp

```diff
@@ -81,56 +80,54 @@ class ForceTorqueSensor : public SemanticComponentInterface<geometry_msgs::msg::
     check_and_add_interface(interface_torque_x, 3);
     check_and_add_interface(interface_torque_y, 4);
     check_and_add_interface(interface_torque_z, 5);
-
-    // Set default force and torque values to NaN
-    std::fill(forces_.begin(), forces_.end(), std::numeric_limits<double>::quiet_NaN());
-    std::fill(torques_.begin(), torques_.end(), std::numeric_limits<double>::quiet_NaN());
   }
 
-  virtual ~ForceTorqueSensor() = default;
-
   /// Return forces.
   /**
    * Return forces of a FTS.
    *
-   * \return array of size 3 with force values.
+   * \return array of size 3 with force values (x, y, z).
    */
   std::array<double, 3> get_forces()
   {
-    size_t interface_counter = 0;
-    for (size_t i = 0; i < 3; ++i)
+    std::array<double, 3> forces{
+      {double_limits::quiet_NaN(), double_limits::quiet_NaN(), double_limits::quiet_NaN()}};
+    size_t interface_counter{0};
+    for (size_t i{0}; i < forces.size(); ++i)
```

Comment:
```suggestion
    for (auto i = 0u; i < forces.size(); ++i)
```
this is the style for these loops that we found least outrageous & compliant with types